syntax = "proto3";

package sbai_geometry_protos;

import "sbai_geometry_protos/transform.proto";
import "sbai_std_protos/header.proto";

// This expresses a transform from coordinate frame header.frame_id
// to the coordinate frame child_frame_id at the time of header.stamp
//
// This message is mostly used by the
// <a href="https://index.ros.org/p/tf2/">tf2</a> package.
// See its documentation for more information.
//
// The child_frame_id is necessary in addition to the frame_id
// in the Header to communicate the full reference for the transform
// in a self contained message.

message TransformStamped {
    sbai_std_protos.Header header = 1;
    string child_frame_id = 2;
    sbai_geometry_protos.Transform transform = 3;
}
